hack/update-hack-tools.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+# Explicitly opt into go modules, even though we're inside a GOPATH directory
+export GO111MODULE=on
+
+# Detect problematic GOPROXY settings that prevent lookup of dependencies
+if [[ "${GOPROXY:-}" == "off" ]]; then
+  kube::log::error "Cannot run hack/update-vendor.sh with \$GOPROXY=off"
+  exit 1
+fi
+
+kube::golang::verify_go_version
+
+pushd "${KUBE_ROOT}/hack/tools" >/dev/null
+  echo "=== tidying go.mod/go.sum in hack/tools"
+  go mod edit -fmt
+  go mod download
+  go mod tidy
+
+  if [[ "${CALLED_FROM_VERIFY:-""}" == "1" ]]; then
+    go clean -cache -modcache -r
```

Comment:
I didn't expect force clearing the modcache... why was this needed?

Reply:
in the verify-hack-tools script, we create a temp directory and use that for the go path etc, i was unable to clean it up unless we cleared the modcache because of https://github.com/golang/go/issues/27161

Reply:
if everything in update-hack-tools uses gomodule mode, you shouldn't need to set a fake GOPATH

Reply:
that means you can just copy the hack/tools dir (don't need the k8s.io/kubernetes parent dir structure, and can stop overriding GOPATH

Reply:
good point, let me try that